_source/_posts/2022-09-01-react-native-spring-boot-jhipster.adoc

```diff
@@ -0,0 +1,233 @@
+
+  adb reverse tcp:8080 tcp:8080 && adb reverse tcp:9080 tcp:9080
+
+. Refresh your app in Simulator using kbd:[{commandKey} + R] or by hitting kbd:[R] twice in Android
+
+== Use Auth0 for Identity
+
+. Log in to your Auth0 account (or https://auth0.com/signup[sign up] if you don't have an account). You should have a unique domain like `dev-xxx.us.auth0.com`.
+
+. Press the *Create Application* button in https://manage.auth0.com/#/applications[Applications section]. Use a name like `JHipster Baby!`, select `Regular Web Applications`, and click *Create*.
+
+. Switch to the *Settings* tab and configure your application settings:
++
+- Allowed Callback URLs: `\http://localhost:8080/login/oauth2/code/oidc`
+- Allowed Logout URLs: `\http://localhost:8080/`
+
+. Scroll to the bottom and click *Save Changes*.
+
+. In the https://manage.auth0.com/#/roles[roles] section, create new roles named `ROLE_ADMIN` and `ROLE_USER`.
+
+. Create a new user account in the https://manage.auth0.com/#/users[users] section. Click on the *Role* tab to assign the roles you just created to the new account.
++
+_Make sure your new user's email is verified before attempting to log in!_
+
+. Next, head to **Actions** > **Flows** and select **Login**. Create a new action named `Add Roles` and use the default trigger and runtime. Change the `onExecutePostLogin` handler to be as follows:
++
+[source,js]
+----
+exports.onExecutePostLogin = async (event, api) => {
+  const namespace = 'https://www.jhipster.tech';
+  if (event.authorization) {
+    api.idToken.setCustomClaim('preferred_username', event.user.email);
+    api.idToken.setCustomClaim(`${namespace}/roles`, event.authorization.roles);
+    api.accessToken.setCustomClaim(`${namespace}/roles`, event.authorization.roles);
+  }
+}
+----
++
+This code is adding the user's roles to a custom claim (prefixed with `https://www.jhipster.tech/roles`). This claim is mapped to Spring Security authorities in `SecurityUtils.java`.
+
+. Select **Deploy** and drag the `Add Roles` action to your Login flow.
+
+. Create a `backend/.auth0.env` file and populate it with your Auth0 settings.
++
+[source,shell]
+----
+export SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_OIDC_ISSUER_URI=https://<your-auth0-domain>/
+export SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_ID=<your-client-id>
+export SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET=<your-client-secret>
+export JHIPSTER_SECURITY_OAUTH2_AUDIENCE=https://<your-auth0-domain>/api/v2/
+----
++
+You can use the default `Auth0 Management API` audience value from the *Applications* > *API* > *API Audience* field. You can also define your own custom API and use the identifier as the API audience.
+
+=== Create a Native OIDC App
+
+. For React Native, create a *Native* app and add the following Allowed Callback URLs:
+
+  http://localhost:19006/,https://auth.expo.io/@mraible/Flickr2,exp://192.168.7.222:19000
```

Comment:
```suggestion
  http://localhost:19006/,https://auth.expo.io/@mraible/Flickr2,exp://192.168.7.222:19000
```
Should mention that 192.168.7.222 is the local IP address and instructions on how to find its value.